### PR TITLE
Add new display options.

### PR DESCRIPTION
**Pull request by SamInPgh**
Add options to 1) display the release year with Albums and 2) display Artist Albums sorted by release date, either ascending or descending, in addition to alphabetically.

### CODE REVIEW

File: Plugin.pm [hunk 1]
```diff
@@ -467,18 +467,22 @@ sub QobuzArtist {
...
+						return $b->{tracks_count} <=> $a->{tracks_count};
+					}
 
-				return $a->{released_at}*1 <=> $b->{released_at}*1 if $sortByDate;
-				return lc($a->{title}) cmp lc($b->{title});
-
+					return lc($a->{title}) cmp lc($b->{title});
+				}
+				
 			} @{$artist->{albums}->{items} || []} ];
```
Comment (michaelherger) [inline Plugin.pm:486]: That's certainly doing the job. 

If we were picky about performance, or dealing with large lists, you could optimize it: in its current state the condition for `$sortByDate` would be evaluated for each and every item. That's probably not much of a problem, as hardly any artist would have more than a few albums (besides Frank Zappa, who's got something between 100-200 on Spotify, IIRC...). But you could evaluate it outside the loop, then create an anonymous sub to execute the sorting comparison. Something like:


```perl
my $sorter;

if ($sortByDate) {
   if ($sortByDate == 1) {
      $sorter = sub { $b->{released_at}*1 <=> $a->{released_at} };
   } else {
      $sorter = sub { $a->{released_at}*1 <=> $b->{released_at} };
   }
}
else {
   $sorter = sub {
      if ( ($a->{tracks_count} >= 4 ...) {
         return $b->{tracks_count} <=> $a->{tracks_count};
      }

      return lc($a->{title})...;   
   };
}

$artist->{albums}->{items} = [ sort $sorter @{$artist->{albums}->{items} || []} ];
```

I guess you get the idea. But as I said: that's just an idea, not really needed here (nor fully implemented or tested).

File: Plugin.pm [hunk 2]
```diff
@@ -1124,7 +1129,8 @@ sub _albumItem {
 
 	if ($albumName) {
 		$item->{line1} = $albumName;
-		$item->{line2} = $artist;
+		$item->{line2} = $artist . ($albumYear ? ' (' . $albumYear . ')' : '');
+		$item->{name} .= $albumYear ? "\n(" . $albumYear . ')' : '';
```
Comment (michaelherger) [inline Plugin.pm:1133]: Which UI would respect that line feed ("`\n`")? If it's always respected, I'd put the year without the brackets, as they look odd, alone on a single line:

```
Greatest Album  - Favorite Artist
(1999)
```

**Review by michaelherger — COMMENTED** [review]
Good job! 

I left a comment "for educational purposes". And was wondering about the year in brackets. But if you like like that, I can certainly live with it.

File: Plugin.pm (hunk 1 above)
Comment (SamInPgh) [reply Plugin.pm:486]: Good point. I'm usually big on code efficiency but this is my first experience with sorting in Perl and I admit there was some cutting and pasting going on here.  ;-)

File: Plugin.pm (hunk 2 above)
Comment (SamInPgh) [reply Plugin.pm:1133]: As far as I can tell, only the native LMS skins use the 'name' field to display the album information. All the other clients use 'line1' and 'line2'. As mentioned on the forum thread, I used the brackets on the year to be consistent with how the LMS skins display the release year with library albums when the Settings->Interface option to do so is specified.

**Pull request merged**